File: front-end/BusinessSite/src/app/model/Introduction.ts
import { ExternalUrl } from './ExternalUrl';

export class Introduction {
    public title: string;
    public content: string;
    public posterUrl: string;
    public posterDescription: string;
    public version: number;

    public posterToUpload: File;
    public posterPreview: string;

    public externalUrls: Array<ExternalUrl>;
}

File: back-end/Abstractions/Models/Introduction.cs
using System.Diagnostics.CodeAnalysis;

namespace Abstractions.Models
{
    [ExcludeFromCodeCoverage]
    /// <summary> Introduction for the index page </summary>
    public class Introduction
    {
        /// <summary> Content </summary>
        public string Content { get; set; } = string.Empty;

        /// <summary> Poster description </summary>
        public string PosterDescription { get; set; } = string.Empty;

        /// <summary> Poster URL </summary>
        public string PosterUrl { get; set; } = string.Empty;

        /// <summary> Introduction title </summary>
        public string Title { get; set; } = string.Empty;

        /// <summary> Entity version </summary>
        public long Version { get; set; }

        /// <summary> External URLs </summary>
        public IEnumerable<ExternalUrl> ExternalUrls { get; set; } = Enumerable.Empty<ExternalUrl>();
    }
}
